website/src/pages/dictionary/index.js

```diff
@@ -185,37 +196,63 @@ function DataDictionary() {
-            let tierBool = false;
-            let attributeBool = false;
-
-            if (tier === NO_ACTIVE_FILTER && attribute === NO_ACTIVE_FILTER) return true;
-
-            if (
-              (tier === TAG_TYPES.id && primaryId) ||
-              (tier === TAG_TYPES.core && core) ||
-              (tier === TAG_TYPES.extended && !core && !primaryId) ||
-              tier === '' ||
-              tier === NO_ACTIVE_FILTER
-            ) {
-              tierBool = true;
-            }
-
-            if (
-              (attribute === TAG_TYPES.conditional && Boolean(dependsOn)) ||
-              (attribute === TAG_TYPES.required && required) ||
-              attribute === '' ||
-              attribute === NO_ACTIVE_FILTER
-            ) {
-              attributeBool = true;
-            }
-
-            return tierBool && attributeBool;
-          });
+          const filteredFields = schema.fields
+            /**
+             * !!!!!!!!!!!!!!!!!!!!!!
+             * Remove this after demo
+             * !!!!!!!!!!!!!!!!!!!!!!
+             */
+            .map((fields, i) => {
+              if (i === 2) {
+                return { ...fields, ...{ compare: 'addition' } };
+              } else if (i === 3) {
+                return { ...fields, ...{ compare: 'deletion' } };
+              } else if (i === 7) {
+                return { ...fields, ...{ compare: 'update' } };
+              } else {
+                return fields;
+              }
+            })
+            .filter((field) => {
+              const meta = get(field, 'meta', {});
+              const { primaryId = false, core = false, dependsOn = false } = meta;
+              const required = get(field, 'restrictions.required', false);
+
+              let tierBool = false;
+              let attributeBool = false;
+
+              if (tier === NO_ACTIVE_FILTER && attribute === NO_ACTIVE_FILTER) return true;
+
+              if (
+                (tier === TAG_TYPES.id && primaryId) ||
+                (tier === TAG_TYPES.core && core) ||
+                (tier === TAG_TYPES.extended && !core && !primaryId) ||
+                tier === '' ||
+                tier === NO_ACTIVE_FILTER
+              ) {
+                tierBool = true;
+              }
+
+              if (
+                (attribute === TAG_TYPES.conditional && Boolean(dependsOn)) ||
+                (attribute === TAG_TYPES.required && required) ||
+                attribute === '' ||
+                attribute === NO_ACTIVE_FILTER
+              ) {
+                attributeBool = true;
+              }
+
+              const compareVal = get(field, 'compare', '');
+              const { additionIsActive, deletionIsActive, updateIsActive } = compareFilters;
+
+              const comparisonBool =
+                (additionIsActive && compareVal === 'addition') ||
+                (deletionIsActive && compareVal === 'deletion') ||
+                (updateIsActive && compareVal === 'update');
```

Comment:
there seems to be some scattered "language" that all seem to refer to the same things... `additionIsActive`, `addition` and `ADDITIONS` are all different manifestation of the same concept. Any chance we can reduce this variant? something like making `compareFilterTypes` something like `{ ADDITION: "ADDITION" ... }` maybe?